libmuscle/python/libmuscle/test/test_mmsf_validator.py

```diff
@@ -0,0 +1,154 @@
+from typing import Any
+from unittest.mock import Mock
+
+import pytest
+from ymmsl import Operator, Reference
+
+from libmuscle.port_manager import PortManager
+from libmuscle.mmsf_validator import MMSFValidator
+
+
+# For testing purposes we monkeypatch _logger.warning so it raises the following
```

Comment:
But pytest has the standard caplog fixture for this, and I think we use it elsewhere. This is a bit of a hack instead. Can this not be done with caplog? Note that there's a `caplog.clear()`, so a function that takes caplog from the test and checks if there's a warning-level message, then clears it, should work I think?